src/sources/aws_s3/sqs.rs

```diff
@@ -0,0 +1,543 @@
+                bucket: s3_event.s3.bucket.name.clone(),
+                key: s3_event.s3.object.key.clone(),
+                ..Default::default()
+            })
+            .await
+            .context(GetObject {
+                bucket: s3_event.s3.bucket.name.clone(),
+                key: s3_event.s3.object.key.clone(),
+            })?;
+
+        let metadata = object.metadata;
+        let timestamp = object
+            .last_modified
+            .and_then(|t| {
+                DateTime::parse_from_rfc2822(&t)
+                    .map(|ts| Utc.timestamp(ts.timestamp(), ts.timestamp_subsec_nanos()))
+                    .ok()
+            })
+            .unwrap_or_else(Utc::now);
+
+        match object.body {
+            Some(body) => {
+                let object_reader = super::s3_object_decoder(
+                    self.compression,
+                    &s3_event.s3.object.key,
+                    object.content_encoding.as_deref(),
+                    object.content_type.as_deref(),
+                    body,
+                );
+
+                // Record the read error saw to propagate up later so we avoid ack'ing the SQS
+                // message
+                //
+                // String is used as we cannot take clone std::io::Error to take ownership in
+                // closure
+                //
+                // FramedRead likely stops when it gets an i/o error but I found it more clear to
+                // show that we `take_while` there hasn't been an error
+                //
+                // This can result in objects being partially processed before an error, but we
+                // prefer duplicate lines over message loss. Future work could include recording
+                // the offset of the object that has been read, but this would only be relevant in
+                // the case that the same vector instance processes the same message.
+                let mut read_error: Option<std::io::Error> = None;
+                let lines: Box<dyn Stream<Item = Bytes> + Send + Unpin> = Box::new(
+                    FramedRead::new(object_reader, BytesDelimitedCodec::new(b'\n'))
+                        .map(|res| {
+                            res.map_err(|err| {
+                                read_error = Some(err);
+                                ()
+                            })
+                            .ok()
+                        })
+                        .take_while(|res| futures::future::ready(res.is_some()))
+                        .map(|r| r.unwrap()), // validated by take_while
+                );
+
+                let lines = match &self.multiline {
+                    Some(config) => Box::new(
+                        LineAgg::new(
+                            lines.map(|line| ((), line, ())),
+                            line_agg::Logic::new(config.clone()),
+                        )
+                        .map(|(_src, line, _context)| line),
```

Comment:
```suggestion
                        .map(|(_, line, _)| line),
```

Reply:
I actually like the `_src` and `_context` as it tells me what these variables would be.

Reply:
I guess so. I'm used to having my editor show me that information when I want to know, and since we're dropping the values here, we "don't care what they are", so the extra naming seems like noise to me.

But it's a minor nit, and it's fine to leave them in 👍

Reply:
True, I probably wouldn't lean on them as much if I configured `vim` better.

Though, it is still nice when browsing the code in Github or other "dumb" editors.